goaviatrix/smart_group.go

```diff
@@ -116,13 +187,8 @@ func (c *Client) CreateSmartGroup(ctx context.Context, smartGroup *SmartGroup) (
 	endpoint := "app-domains"
 	form := makeSmartGroupForm(smartGroup)
 
-	type SmartGroupResp struct {
-		UUID string `json:"uuid"`
-	}
-
 	var data SmartGroupResp
-	err := c.PostAPIContext25(ctx, &data, endpoint, form)
-	if err != nil {
+	if err := c.PostAPIContext25(ctx, &data, endpoint, form); err != nil {
 		return "", err
 	}
 	return data.UUID, nil
```

Comment:
Since you aren't using a pointer, `data.UUID` should be empty string already, right?

If so, this would trim it down further, but may be confusing. Feel free to disregard.

```suggestion
	err := c.PostAPIContext25(ctx, &data, endpoint, form)
	return data.UUID, err
```